modules/database/export.bal

```diff
@@ -272,40 +279,89 @@ public isolated transactional function getTimelineStepLimit(int experimentId) re
+        add = true;
+        if config.restriction == "STEPS" && timelineStepDb.sequence == stepList[counter] {
+            // filter steps with stepList 
+            if timelineStepDb.sequence == stepList[counter] {
+                counter += 1;
+            } else {
+                add = false;
+            }
         }
-        // retrieve associated substeps with their substep data
-        timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId);
-        timelineSteps.push(timelineStepExport);
-        foreach var substep in timelineStepExport.timelineSubsteps {
-            foreach var substepData in substep.substepDataList {
-                dataIdList.push(substepData.dataId);
+        if add {
+            TimelineStepExport timelineStepExport = check castToTimelineStepExport(timelineStepDb);
+            if config.restriction != "LOGS" {
+                // retrieve associated step data
+                timelineStepExport.stepDataList = check getTimelineStepDataList(timelineStepDb.stepId);
+                foreach var stepData in timelineStepExport.stepDataList {
+                    dataIdList.push(stepData.dataId);
+                }
+            } else {
+                // don't need data for restriction "LOGS"
+                timelineStepExport.stepDataList = [];
             }
+            // retrieve associated substeps with their substep data
+            timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId, config);
+            timelineSteps.push(timelineStepExport);
+            if config.restriction != "LOGS" {
+                // retrieve associated substep data
+                foreach var substep in timelineStepExport.timelineSubsteps {
+                    foreach var substepData in substep.substepDataList {
+                        dataIdList.push(substepData.dataId);
+                    }
+                }
+            } // else don't need data
         }
     }
 
-    // ignore duplicates in dataIdList and create experiment data list
-    int[] sortedDataIdList = dataIdList.sort();
-    int tmp = -1;
-    foreach int dataId in sortedDataIdList {
-        if dataId != tmp {
-            tmp = dataId;
-            ExperimentDataExport experimentData = check getExperimentDataExport(experimentId, dataId);
-            experimentDataList.push(experimentData);
+    if config.restriction != "LOGS" {
+        // ignore duplicates in dataIdList and create experiment data list
+        int[] sortedDataIdList = dataIdList.sort();
+        int tmp = -1;
+        foreach int dataId in sortedDataIdList {
+            if dataId != tmp {
+                tmp = dataId;
+                ExperimentDataExport experimentData = check getExperimentDataExport(experimentId, dataId);
+                experimentDataList.push(experimentData);
+            }
+        }
+    } // else don't need data
+
+    if config.restriction == "DATA" {
```

Comment:
Would it be better to use a completely different method to export only the data for this case? Data can easily be filtered by experiment if I recall correctly.

Reply:
Yes, correct. Also more efficient that way.